Sources/TensorFlow/Core/Tensor.swift

```diff
@@ -184,24 +186,26 @@ extension Tensor where Scalar: TensorFlowFloatingPoint {
 public extension Tensor {
     /// Creates a 0-D tensor from a scalar value.
     @inlinable
-    @differentiable(vjp: _vjpScalarInit where Scalar: TensorFlowFloatingPoint)
+    @differentiable(where Scalar: TensorFlowFloatingPoint)
     init(_ value: Scalar, on device: Device = Device.getDefault) {
         self.init(shape: [], scalars: [value], on: device)
     }
 }
 
 internal extension Tensor where Scalar: TensorFlowFloatingPoint {
     @inlinable
-    static func _vjpScalarInit(_ value: __owned Scalar, on device: Device = Device.getDefault
-    ) -> (Tensor, (Tensor) -> Scalar) {
+    @derivative(of: init(_:on:))
+    static func _vjpScalarInit(_ value: __owned Scalar, on device: Device = Device.getDefault) -> (
```

Comment:
@dan-zheng Perhaps it would make sense to rename _vjpXXX functions to _derivativeXXX?

Reply:
I don't feel strongly! The best name is `func _` (a private anonymous function), which is used in the [differentiable programming manifesto](http://bit.ly/swift-autodiff-manifesto) but not yet implemented.

When linear functions and transposition are done, users will register only primitive JVP functions returning a `@differentiable(linear)` differential function, and primitive VJP functions will be removed. At that point, `@derivative` functions will unambiguously be JVP functions, so `derivativeXXX` names become appropriate and `vjpXXX` names become obsolete.